src/main/java/org/jenkinsci/plugins/GithubAuthorizationStrategy.java

```diff
@@ -55,6 +55,7 @@ public class GithubAuthorizationStrategy extends AuthorizationStrategy {
 
     @DataBoundConstructor
     public GithubAuthorizationStrategy(String adminUserNames,
+            String agentUserName,
```

Comment:
In order to avoid a breaking change, I believe this will need to be optional. Then it can default to a blank string. For example users of the Jenkins [configuration as code plugin](https://plugins.jenkins.io/configuration-as-code/) would not have `agentUserName: ""` in their yaml, and so if they pull the `latest` of this plugin, their Jenkins instance will not launch.

The way that I went about this was to leave it out of the constructor and instead define a getter and setter:
https://github.com/jenkinsci/github-oauth-plugin/pull/246/files#diff-fb9af14b2ea3f59e6b202cb37bfedc25d10247c1cc99b9220830b52e51d7b353R149-R164

Reply:
Great idea, thank you! :)